constants/ast.js

```diff
@@ -0,0 +1,12 @@
+let i = 0;
+
+const ast_types = {
+    tag         : i++,
```

Comment:
Is there any reason not to just hardcode the values?

Reply:
I've seen this pattern be used in many repos, it's just to make adding new nodes simpler.